bioscrape/inference_setup.py

```diff
@@ -302,9 +270,26 @@ def extract_data(self, exp_data):
                 all_initial_conditions = [self.initial_conditions]*N
             elif type(self.initial_conditions) is list:
                 if len(self.initial_conditions) != N:
-                    raise ValueError('For a list of initial conditions, each item must be a dictionary and the length of the list must be the same as the number of trajectories.')
+                    raise ValueError('For a list of initial conditions,'
+                                     'each item must be a dictionary and'
+                                     'the length of the list must be the'
+                                     'same as the number of trajectories.')
                 all_initial_conditions = self.initial_conditions
             self.initial_conditions = all_initial_conditions
+            # Create parameter conditions as required
+            if type(self.parameter_conditions) is dict:
```

Comment:
Shouldn't these checks be in set_parameter_conditions?

Reply:
Since I am also checking the size of data list/dataframe passed to make sure the lengths of condition lists are correct, I am doing these checks later in `prepare_...` functions.